transform-plugins/src/main/java/co/cask/hydrator/plugin/ValueMapper.java

```diff
@@ -0,0 +1,251 @@
+import co.cask.cdap.etl.api.LookupTableConfig;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.Transform;
+import co.cask.cdap.etl.api.TransformContext;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Transforms records using custom mapping provided by the config.
+ */
+@Plugin(type = Transform.PLUGIN_TYPE)
+@Name("ValueMapper")
+@Description("Maps and converts record values using a mapping dataset")
+public class ValueMapper extends Transform<StructuredRecord, StructuredRecord> {
+
+  private final Config config;
+  private final Map<Schema, Schema> schemaCache = new HashMap<>();
+  private static final Map<String, ValueMapping> mappingValues = new HashMap<>();
+
+  //for unit tests, otherwise config is injected by plugin framework.
+  public ValueMapper(Config config) {
+    this.config = config;
+    this.config.parseConfiguration();
+  }
+
+  /**
+   * Configuration for the ValueMapper transform.
+   */
+  public static class Config extends PluginConfig {
+
+    @Name("mapping")
+    @Description("Specify the source and target field mapping and lookup dataset name." +
+            "Format is <source-field>:<lookup-table-name>:<target-field>" +
+            "[,<source-field>:<lookup-table-name>:<target-field>]*" +
+            "Source and target field can only of type string." +
+            "For eg: lang_code:language_code_lookup:lang_desc,country_code:country_lookup:country_name")
+    private final String mapping;
+
+    @Name("defaults")
+    @Description("Specify the defaults for source fields if the lookup does not exist or inputs are NULL or EMPTY. " +
+            "Format is <source-field>:<default-value>[,<source-field>:<default-value>]*" +
+            "lang_code:English,country_code:Britain")
+    private final String defaults;
+
+    public Config(String mapping, String defaults) {
+      this.mapping = mapping;
+      this.defaults = defaults;
+    }
+
+    /**
+     * This method is meant to parse input configuration.
+     * It is required to use in configurePiperline as well as transform method.
+     * Hence this is implemented as a part of Config class to set configuration once and make it available for
+     * subsequent methods.
+     */
+    private void parseConfiguration() {
+      Map<String, String> defaultsMapping = new HashMap<>();
+      if (!defaults.isEmpty()) {
+        String[] defaultsList = this.defaults.split(",");
+        for (String defaultValue : defaultsList) {
+          String[] defaultsArray = defaultValue.split(":");
+          defaultsMapping.put(defaultsArray[0], defaultsArray[1]);
+        }
+      }
+      String[] mappingArray = this.mapping.split(",");
+      for (String mapping : mappingArray) {
+        String[] mappingValueArray = mapping.split(":");
+        ValueMapping valueMapping = new ValueMapping();
+        valueMapping.setTargetField(mappingValueArray[2]);
```

Comment:
Need some clarifications. As, validations should be done in configurePipeline(), but currently we are parsing the inputs in Config class , so is it ok if i validate the items and throw the exceptions from Config class itself?